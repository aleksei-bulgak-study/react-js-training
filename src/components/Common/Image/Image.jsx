import React from 'react';
import PropTypes from 'prop-types';

const Image = ({ src, alt, className }) => (
  <img className={className} src={src} alt={alt} />
);

Image.propTypes = {
  src: PropTypes.string.isRequired,
  alt: PropTypes.string,
  className: PropTypes.string,
};

Image.defaultProps = {
  alt: '',
  className: '',
};

export default Image;
