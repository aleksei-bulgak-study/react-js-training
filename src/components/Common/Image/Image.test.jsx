import React from 'react';
import { shallow } from 'enzyme';
import Image from './Image';

describe('when Image', () => {
  test('with src and alt provided then cmponent will be rendered', () => {
    const src = "image path";
    const image = shallow(<Image src={src} alt="testDescription" className="testImage"/>);

    expect(image.find('.testImage').prop('src')).toEqual(src);
    expect(image.find('.testImage').prop('alt')).toEqual("testDescription");
    expect(image).toMatchSnapshot();
  });
});
