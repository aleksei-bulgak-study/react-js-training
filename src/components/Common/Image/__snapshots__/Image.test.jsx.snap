// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`when Image with src and alt provided then cmponent will be rendered 1`] = `ShallowWrapper {}`;
